raven/processes/regionalization_methods.py

```diff
@@ -0,0 +1,326 @@
+ 
+    Qsim=np.delete(Qsim,list(range(Qs.shape[0],Qsim.shape[0])),axis=0);
+
+    
+    return Qsim
+
+
+def distance_MLR_regionalization(ncfilename,start_date, end_date, latitude, longitude, elevation, area,
+                                 hydrological_model, inputs_file,
+                                 number_donors,
+                                 ungauged_properties,
+                                 model_parameters,
+                                 gauged_properties,
+                                 properties_to_use):
+    """
+    distance_MLR_regionalization
+
+    INPUTS:
+        hydrological_model -- String of the selected hydrological model
+        inputs_file -- The ungauged catchment inputs file to work with
+        number_donors -- The number of gauged catchments to use
+        ungauged_properties -- Properties of the ungauged catchment
+        model_parameters -- Selected hydrological model calibrated parameters
+        gauged_properties -- Properties of all gauged catchments available
+        properties_to_use -- The catchments' properties to use
+
+    Function to compute distance MLR regionalization method.
+    """
+
+    MLR_parameters, r_squared = MLR(ungauged_properties,
+                                    model_parameters,
+                                    gauged_properties,
+                                    properties_to_use)
+
+    model_parameters = model_parameters.values[:, 2:]
+
+    for i in range(0, r_squared.shape[0]):
+        # if we have an R2 > 0.5 then we consider this to be a better estimator
+        if r_squared[i] > 0.5:
+            model_parameters[:, i] = MLR_parameters[i]
+
+    nc_file = Dataset(inputs_file[0], 'r')
+    nDays=nc_file.variables["rain"][:].shape[0]
+    Qsim = np.zeros(shape=(nDays,number_donors))
+    
+
+    for j in range(0, number_donors):
+        
+       params=",".join(map(str,model_parameters[j,:])) 
+
+       
+       if hydrological_model == "GR4JCN":
+        
+           Qs=runLocalGR4JCN(params,ncfilename,start_date, end_date, latitude, longitude, elevation, area)
+           Qsim[0:Qs.shape[0],j]=Qs.flatten() 
+        
+       elif hydrological_model == "HMETS":
+        
+           Qs=runLocalHMETS(params,ncfilename,start_date, end_date, latitude, longitude, elevation, area )
+           Qsim[0:Qs.shape[0],j]=Qs.flatten()
+            
+       elif hydrological_model == "MOHYSE":
+           params=",".join(map(str,model_parameters[j,0:7]))
+           hrus=",".join(map(str,model_parameters[j,8:9]))
+           
+           Qs=runLocalMOHYSE(params,hrus,ncfilename,start_date, end_date, latitude, longitude, elevation, area )
+           Qsim[0:Qs.shape[0],j]=Qs.flatten()
+        
+       else:
+           print("The hydrological model selected is not supported for \
+                 regionalization")
+
+    return Qsim
+
+
+def runLocalHMETS(params,ncfilename,start_date, end_date, latitude, longitude, elevation, area ):

```

Comment:
just saw your docs, indeed it'll be easier this way... I spent a lot of time making this WPS-WPS connection work.

Reply:
Sorry about that. I'll see if I can update the run method to flatten the argument list (pass start_date directly instead of inside the rvi dictionary.).